controllers/apps/apimanager_status_reconciler.go

```diff
@@ -55,6 +56,10 @@ func (s *APIManagerStatusReconciler) Reconcile() (reconcile.Result, error) {
 	if equalStatus && apiManagerAvailable {
 		// Steady state
 		s.logger.V(1).Info("Status was not updated")
+
+		// 3scale 2.14 -> 2.15
+		err = upgrade.DeleteImageStreams(s.apimanagerResource.Namespace, s.Client())
```

Comment:
hey @carlkyrillos why did you placed it here in status reconciler?

WDYT if we move it https://github.com/3scale/3scale-operator/pull/929/files#diff-6ebc088c3986da1f5f483589ce7d214c2047c71ce02c8a7c697bc216563398e7R22

Or to the apimanager reconciler (this can be the last step in apim reconciliation however, I'm in favor of moving the migration logic to migration function)

Reply:
Good call, I moved it to the APIManager controller